src/Umbraco.Infrastructure/PropertyEditors/RichTextEditorPastedImages.cs

```diff
@@ -56,149 +67,301 @@ public sealed class RichTextEditorPastedImages
     }
 
     /// <summary>
-    ///     Used by the RTE (and grid RTE) for drag/drop/persisting images
+    ///     Used by the RTE (and grid RTE) for converting inline base64 images to Media items.
     /// </summary>
-    public string FindAndPersistPastedTempImages(string html, Guid mediaParentFolder, int userId, IImageUrlGenerator imageUrlGenerator)
+    /// <param name="html">HTML from the Rich Text Editor property editor.</param>
+    /// <param name="mediaParentFolder"></param>
+    /// <param name="userId"></param>
+    /// <returns>Formatted HTML.</returns>
+    /// <exception cref="NotSupportedException">Thrown if image extension is not allowed</exception>
+    internal string FindAndPersistEmbeddedImages(string html, Guid mediaParentFolder, int userId)
     {
         // Find all img's that has data-tmpimg attribute
         // Use HTML Agility Pack - https://html-agility-pack.net
         var htmlDoc = new HtmlDocument();
         htmlDoc.LoadHtml(html);
 
-        HtmlNodeCollection? tmpImages = htmlDoc.DocumentNode.SelectNodes($"//img[@{TemporaryImageDataAttribute}]");
-        if (tmpImages == null || tmpImages.Count == 0)
+        HtmlNodeCollection? imagesWithDataUris = htmlDoc.DocumentNode.SelectNodes("//img");
+        if (imagesWithDataUris == null || imagesWithDataUris.Count == 0)
         {
             return html;
         }
 
-        // An array to contain a list of URLs that
-        // we have already processed to avoid dupes
-        var uploadedImages = new Dictionary<string, GuidUdi>();
-
-
-        foreach (HtmlNode? img in tmpImages)
+        foreach (HtmlNode? img in imagesWithDataUris)
         {
-            // The data attribute contains the path to the tmp img to persist as a media item
-            var tmpImgPath = img.GetAttributeValue(TemporaryImageDataAttribute, string.Empty);
+            var srcValue = img.GetAttributeValue("src", string.Empty);
 
-            if (string.IsNullOrEmpty(tmpImgPath))
+            // Ignore src-less images
+            if (string.IsNullOrEmpty(srcValue))
             {
                 continue;
             }
 
+            // Take only images that have a "data:image" uri into consideration
+            if (!srcValue.StartsWith("data:image"))
+            {
+                continue;
+            }
 
-            var absoluteTempImagePath = Path.GetFullPath(_hostingEnvironment.MapPathContentRoot(tmpImgPath));
+            // Create tmp image by scanning the srcValue
+            // the value will look like "data:image/jpg;base64,abc" where the first part
+            // is the mimetype and the second (after the comma) is the image blob
+            Match dataUriInfo = Regex.Match(srcValue, @"^data:\w+\/(?<ext>\w+)[\w\+]*?;(?<encoding>\w+),(?<data>.+)$");
 
-            if (IsValidPath(absoluteTempImagePath) == false)
+            // If it turns up false, it was probably a false-positive and we can't do anything with it
+            if (dataUriInfo.Success == false)
```

Comment:
Lets change this to pattern matching while we're at it 🔥
```suggestion
            if (dataUriInfo.Success is false)
```